src/box/vy_scheduler.c

```diff
@@ -1339,8 +1339,7 @@ vy_task_dump_new(struct vy_scheduler *scheduler, struct vy_worker *worker,
 	}
 
 	/* Rotate the active tree if it needs to be dumped. */
-	if (lsm->mem->generation == scheduler->dump_generation &&
```

Comment:
Now there's additional cond: `lsm->mem->space_cache_version != space_cache_version`. I believe it's probably OK, but could it lead to some undesirable side-effects? I mean here we can potentially get mem rotation on compaction..

Reply:
Nothing bad should happen, because we won't include the `vy_mem` into dump unless its generation <= dump generation, see:

https://github.com/tarantool/tarantool/blob/47ad3bc9fc1f0e2fc8658d1e4013ce35694d2bf5/src/box/vy_scheduler.c#L1397-L1402

Moreover, the mem can be rotated by tx write anyway, because there's a time window between bumping dump generation and proceeding to dump.